includes/bfcache-opt-in.php

```diff
@@ -492,3 +498,34 @@ function filter_nocache_headers( $headers ): array {
 	__NAMESPACE__ . '\filter_nocache_headers',
 	1000
 );
+
+
+/**
+ * Conditionally adds the Permissions-Policy header to enable BFCache.
+ *
+ * This function checks if the 'bfcache_block_unload_events' option is enabled.
+ * If it is, the Permissions-Policy header with 'unload=()' is added to the HTTP
+ * response. This prevents 'unload' event listeners from firing, allowing the
+ * browser's Back/Forward Cache (BFCache) to function and improve navigation
+ * performance.
+ *
+ * @param array<string, string>|mixed $headers An array of HTTP headers to be sent.
+ * @return array<string, string> The filtered array of headers.
+ * @since n.e.x.t
+ */
+
+
+function disallow_unload_events_permissions_policy_header( $headers ): array {
```

Comment:
```suggestion
function disallow_unload_events_permissions_policy_header( $headers ): array {
	// Because plugins do bad things.
	if ( ! is_array( $headers ) ) {
		$headers = array();
	}

```

Reply:
See the `filter_nocache_headers()` function for what it does.